src/main/java/net/sf/jabref/exporter/BibtexDatabaseWriter.java

```diff
@@ -0,0 +1,167 @@
+            if (!reformatFile && !bibtexString.hasChanged()) {
+                getWriter().write(bibtexString.getParsedSerialization());
+                return;
+            }
+
+            if (isFirstString) {
+                getWriter().write(Globals.NEWLINE);
+            }
+
+            getWriter().write(STRING_PREFIX + "{" + bibtexString.getName() + StringUtil
+                    .repeatSpaces(maxKeyLength - bibtexString.getName().length()) + " = ");
+            if (bibtexString.getContent().isEmpty()) {
+                getWriter().write("{}");
+            } else {
+                try {
+                    String formatted = new LatexFieldFormatter().format(bibtexString.getContent(), LatexFieldFormatter.BIBTEX_STRING);
+                    getWriter().write(formatted);
+                } catch (IllegalArgumentException ex) {
+                    throw new IllegalArgumentException(
+                            "The # character is not allowed in BibTeX strings unless escaped as in '\\#'.\n" + "Before saving, please edit any strings containing the # character.");
+                }
+            }
+
+            getWriter().write("}" + Globals.NEWLINE);
+        } catch (IOException e) {
+            throw new SaveException(e);
+        }
+    }
+
+    @Override
+    protected void writeEntryTypeDefinition(CustomEntryType customType) throws SaveException {
+        try {
+        getWriter().write(Globals.NEWLINE);
+        getWriter().write(COMMENT_PREFIX + "{");
+        getWriter().write(CustomEntryType.ENTRYTYPE_FLAG);
+        getWriter().write(customType.getName());
+        getWriter().write(": req[");
+        getWriter().write(customType.getRequiredFieldsString());
+        getWriter().write("] opt[");
+        getWriter().write(String.join(";", customType.getOptionalFields()));
+        getWriter().write("]}");
+            getWriter().write(Globals.NEWLINE);
+        } catch (IOException e) {
+            throw new SaveException(e);
+        }
+    }
+
+    @Override
+    protected void writePrelogue(BibDatabaseContext bibDatabaseContext, Charset encoding) throws SaveException {
+        if(encoding == null) {
+            return;
+        }
+
+        // Writes the file encoding information.
+        try {
+            getWriter().write("% ");
+            getWriter().write(Globals.ENCODING_PREFIX + encoding);
+            getWriter().write(Globals.NEWLINE);
+        } catch (IOException e) {
+            throw new SaveException(e);
+        }
+    }
+
+    @Override
+    protected void writeEntry(BibEntry entry, BibDatabaseMode mode, Boolean isReformatFile) throws SaveException {
+        BibEntryWriter bibtexEntryWriter = new BibEntryWriter(new LatexFieldFormatter(), true);
+        try {
+            bibtexEntryWriter.write(entry, getWriter(), mode, isReformatFile);
+        } catch (IOException e) {
+            throw new SaveException(e, entry);
+        }
+    }
+
+    private Writer getWriter() {
```

Comment:
I would suggest making that a field, e.g. Writer writer and use that everywhere instead of getWriter()
Improves readability